pkg/ottl/contexts/internal/ottlcommon/span.go

```diff
@@ -339,6 +339,9 @@ func accessAttributesKey[K SpanContext](mapKey *string) ottl.StandardGetSetter[K
 			return GetMapValue(tCtx.GetSpan().Attributes(), *mapKey), nil
 		},
 		Setter: func(ctx context.Context, tCtx K, val interface{}) error {
+			if ok := checkValuesHomogeneous(val); !ok {
```

Comment:
We should enforce homogeneous arrays for resource, scope, spanevent, datapoint and logs as well. @evan-bradley is that an accurate list for homogeneous arrays and the spec?

Reply:
I believe that should cover everything, yes. According to [the specification](https://opentelemetry.io/docs/reference/specification/common/#attribute), this will apply for attribute values for any telemetry scope. Any `attributes` path in the OTTL should be enforcing this.

Reply:
I add  CheckValuesHomogeneous to more function, but I'm not sure I cover all the cases you mention, could you help check about this? @evan-bradley